en/blog/nixos-remote-development.md

```diff
@@ -0,0 +1,212 @@
+                format = "vfat";
+                mountpoint = "/boot";
+              };
+            };
+            root = {
+              size = "100%";
+              content = {
+                type = "filesystem";
+                format = "ext4";
+                mountpoint = "/";
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+}
+```
+
+The two important attributes here are:
+- `disko.devices.disk.<device-name>.device`: The device name of the disk to partition.
+<!-- Is the boot partition necessary? -->
+- `disko.devices.disk.<device-name>.content.partitions`:
+    - [`boot`](https://en.wikipedia.org/wiki/BIOS_boot_partition)
+    - [`ESP`](https://en.wikipedia.org/wiki/EFI_system_partition)
+    - [`root`](https://en.wikipedia.org/wiki/Root_directory)
+
+### `configuration.nix` as a module
+
+In pre-[[flakes]] world, `configuration.nix` has been the top-level [[modules|module]] specifying entire NixOS configuration. When using [`flake.nix`](#the-flakenix-file), however, we can add the configuration to the `modules` attribute of `nixosSystem` function.
+
+```nix
+# Inside `outputs`
+{
+  nixosConfigurations.office = lib.nixosSystem {
+    # ...
+    modules = [
+      # ...
+      ({ ... }: {
+        # Your `configuration.nix` goes here
+      })
+    ];
+  }
+}
+```
+
+In this module we do the following things:
+- Enable SSH access to the machine.
+- Set a password for the `root` user.
+- Add an SSH key to the `root` user's `authorized_keys` file.
+- Enable GRUB with EFI support.
+<!-- Verify the point below -->
+- Set the `system.stateVersion` to `23.11` to avoid rebuilding the system on every `nixos-rebuild switch`.
+- Import the `disk-config.nix` file that we created [earlier](#disko-module).
+
+## Installing NixOS
+
+Once our configuration is ready, we can deploy it to a remote machine over SSH. This machine must already be running a linux based OS. If it doesn't, boot it using a [live/rescue image](https://en.wikipedia.org/wiki/Live_USB).
+
+>[!info] kexec
+> If a machine has no OS installed on it yet, how do we install an OS without a bootable USB? The answer is, we use the RAM as a bootable disk. We achieve this with [`kexec`](https://en.wikipedia.org/wiki/Kexec) to load the nixos image into RAM. Once loaded, the control is switched from your current OS to the image running on the RAM, this image then partitions the disk and installs the system on your hard drive.
+
+[`nixos-anywhere`](https://github.com/nix-community/nixos-anywhere) is a tool that automates this process for us. To follow along:
+```sh
+git clone https://github.com/juspay/remote-development.git
+cd remote-development
+git checkout a23acb9cb0a51e048096b3e4c8130b979ca0c2fa
+```
+
+The [README](https://github.com/juspay/remote-development/blob/a23acb9cb0a51e048096b3e4c8130b979ca0c2fa/README.md) should help you get started with the installation. Once you are done, you should be able to SSH into the machine.
```

Comment:
@shivaraj-bh I feel like the user is left wondering in the void after this. We should add an asciicast or video maybe.